### ISSUE TITLE
[FEATURE] BTP Terraform Exporter

### ISSUE THREAD

**Issue by lechnerc77** [SAP/terraform-provider-btp#724]
### What area do you want to see improved?

other

### Is your feature request related to a problem? Please describe.

The Terraform Provider for SAP BTP currently supports the import of resources via the standard functionality of Terraform namely `import` blocks ([link](https://developer.hashicorp.com/terraform/language/import)).
This allows our customers to import resources, but the process is quite cumbersome and error-prone.

It would be great to have a dedicated functionality to execute that export in an automated and streamlined way like:

```bash
btpexport --subaccount `name of the subaccount`
```

### Describe the solution you would like

I would like to have a solution that fulfills the following requirements:

- Automatically identify resources to be imported
- Automatically generate the valid Terraform configuration file:
  - The resource should match its state in the remote platform (verified via terraform plan)
  - Defines the cross resource dependencies
- Option to automatically generate the Terraform state file
- Up-to-date support to the provider (Delay between new resources in the Provider and availability in the tool should be minimal)

There will be gaps as some resources might not support the import  or the configuration might not be accessible (like for some service instances). This is acceptable as long as it is logged/tracked by the tool in a machine-readable manner e.g. by a dedicated JSON file that highlights the resources that could not be imported.

### Describe alternatives you have considered

The [Terraformer](https://github.com/GoogleCloudPlatform/terraformer) project initiated by GCP SRE might be an alternative. It is not clear how well maintained this project is and there is no governance described in the repository that outlines contribution and review flows.

### Additional context

As a blueprint for the requirement the [Microsoft Azure Export for Terraform](https://github.com/Azure/aztfexport) CLI can be used, as it covers exactly the same requirements, but is of course restricted to Azure:

- [Documentation](https://learn.microsoft.com/azure/developer/terraform/azure-export-for-terraform/export-terraform-overview)
- [GitHub Repository](https://github.com/Azure/aztfexport)
- [Original project outline and motivation](https://azure.github.io/aztfexport/) - a bit outdated

**Comment by lechnerc77** [comment]
Additional remark: for a end2end story of an export the role collection assignment would be an important part. However, this resource is not importable due to missing functionality to READ the role collection assignment. This gap is something that probably needs further discussion also in the context of  #748

**Comment by lechnerc77** [comment]
This functionality will be implemented

**Comment by ptesny** [comment]
@lechnerc77 
While awaiting the feature to be implemented.

My use case is the internal ArgoCDaaS. I’d like to import/export the ArgoCDaaS service instance binding (different BTP ga and sa) for use with my runtime tf script ( in a different BTP ga/sa);
But this could be applicable to an existing BTP assets like HC,ObjectStore, Postgres, DataSphere, SAC, etc…
Any BTP real estate import example would help.

**Comment by lechnerc77** [comment]
@ptesny : There is a sample that guides you through the import process: https://github.com/SAP-samples/btp-terraform-samples/tree/main/released/import

It does not yet use the automatic generation (experimental) which would you get rid of crafting the configuration manually. Having said that when executing an import with generating the configuration you will probably need some postprocessing to clean up the generated code.

**Comment by se-wo** [comment]
@lechnerc77 you are aware of https://github.com/GoogleCloudPlatform/terraformer already supporting many clouds. Or similar tools like https://github.com/zscaler/zscaler-terraformer

**Comment by lechnerc77** [comment]
@se-wo There are several tools available (vendor-agnostic and vendor specific) that we took a look at. All the tools that we analyzed (one of them was indeed terraformer). The main paatern we saw at the generic ones were: 

-  no well defined governance 
- no well defined contribution model
- outdated/not maintained (like terraformer where the last release was 2023)

Hence we decided to implement a dedicated solution/CLI for SAP BTP. This will land soon (probably in a view weeks) as a first beta version.
